### PR TITLE
EM-104: Update few lines of Argo Workflows UI to allow proper embedding with GreenOps

### PR DESCRIPTION
**Pull request by morzhanov**
## Issue

Update Argo Workflows server in order to restrict iframe embedding on all websites but Greenops UI.
Also, change cookies generation so Greenops UI will be able to access them.

## Solution

- [x] updated cookie handling for sso and client auth
- [x] updated CSP header to allow iframe embedding only on Greenops website

## Testing

Currently, we setting a CSP policy to allow requests from https://localhost:8081/. In future, this should be changed to the Greenops UI production URL.

Steps to build Greenops version of argo-server:

1. build ui:
    
    ```bash
    export STATIC_FILES=true
    make cli
    ```
    
2. build argo-server image:
    
    ```bash
    docker build \
        --platform="linux/amd64" \
        --target argocli \
        --tag greenopsinc/argo-server:3.3.8 .
    ```
    
3. load the image into minikube:
    
    ```bash
    minikube image load greenopsinc/argo-server:3.3.8
    ```
    
4. deploy updated argo-server deployment:
    
    ```bash
    kubectl apply -f ./docs/greenops/argo-server-deployment-example.yaml -n argo
    ```
    

Try to login on the argo workflows side and open pipeline run details page on greenops UI. Iframe should be visible.

Try to embed iframe to some other website. You should see an error in the iframe:
image

### CODE REVIEW

File: docs/greenops/argo-server-deployment-example.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+apiVersion: apps/v1
```
Comment (mihirpandya-greenops) [inline docs/greenops/argo-server-deployment-example.yaml]: Move this to one of the GreenOps folders, we don't want to leave too much in the Argo Workflows repo

File: manifests/base/argo-server/argo-server-deployment.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ spec:
       serviceAccountName: argo-server
       containers:
         - name: argo-server
-          image: quay.io/argoproj/argocli:latest
+          image: greenopsinc/argo-server:3.3.8
```
Comment (mihirpandya-greenops) [inline manifests/base/argo-server/argo-server-deployment.yaml:17]: The server is started via the CLI, so make sure that as a part of the containerization steps that it is the CLI image that is run (there should probably be info on this in the Argo Workflows repo)

File: server/static/static.go [hunk 3]
```diff
@@ -29,9 +29,9 @@ func (s *FilesServer) ServerFiles(w http.ResponseWriter, r *http.Request) {
 		w = &responseRewriter{ResponseWriter: w, old: []byte(`<base href="/">`), new: []byte(fmt.Sprintf(`<base href="%s">`, s.baseHRef))}
 	}
 
-	if s.xframeOpts != "" {
-		w.Header().Set("X-Frame-Options", s.xframeOpts)
-	}
+	// if s.xframeOpts != "" {
```
Comment (mihirpandya-greenops) [inline server/static/static.go:37]: Let's just uncomment this for now, its ok

File: server/static/static.go [hunk 4]
```diff
@@ -45,7 +45,7 @@ func (s *FilesServer) ServerFiles(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// `data:` is need for Monaco editors wiggly red lines
-	w.Header().Set("Content-Security-Policy", "default-src 'self' 'unsafe-inline'; img-src 'self' data:")
+	w.Header().Set("Content-Security-Policy", "default-src 'self' 'unsafe-inline'; img-src 'self'; frame-ancestors https://localhost:8081")
```
Comment (mihirpandya-greenops) [inline server/static/static.go]: Add ENV VAR for custom frame-ancestor

**Comment by morzhanov** [comment]
Update branch with the argo-workflows latest stable version

File: manifests/base/argo-server/argo-server-deployment.yaml (hunk 2 above)
Comment (morzhanov) [reply manifests/base/argo-server/argo-server-deployment.yaml:17]: We're already using cli build for that, so no issues with the current approach. I'll update our repo to the latest stable version and create new docker image.

File: docs/greenops/argo-server-deployment-example.yaml (hunk 1 above)
Comment (morzhanov) [reply docs/greenops/argo-server-deployment-example.yaml]: Moved this file to the greenops repo: https://github.com/GreenOpsInc/greenops-enterprise/pull/60

File: server/static/static.go (hunk 4 above)
Comment (morzhanov) [reply server/static/static.go]: Added env var

**Comment by morzhanov** [comment]
Changed target branch to the `release-3.3` which is the latest stable branch.

**Pull request closed without merging**